src/apps/ociochecklut/main.cpp

```diff
@@ -97,6 +97,7 @@ class ProcessorWrapper
 
 #ifdef OCIO_GPU_ENABLED
     OCIO::OglAppRcPtr m_oglApp;
+    OCIO::GpuShaderDescRcPtr m_shaderDesc;
```

Comment:
I do not seen the need to keep `shaderDesc` as a member variable. What is the problem you try to solve?

Reply:
We only have the title of the commits. Anyway, line 91 is already done at line 69 so it recreates two times the complete GPU fragment shader and resources (`m_oglApp->setShader(m_shaderDesc)` called twice). That seems way overkill. But you found a problem..

Reply:
Sorry, I expected everyone could see the commit message, here it is:

> In OglApp::applyGPU, call to updateImage() update the currently bound GL_TEXTURE_2D
> target, by doing so, the latter sampling of a 1DLUT that was allocated as
> a GL_TEXTURE_2D (because of it's size), will sample the wrong texture.
> 
> Solution proposed is to call OglApp::setShader() again after image was updated.

I agree it's overkill, but ociocheck is not really about performance. I guess we could extend OglApp to have a more lightweight method to set the correct GL state.

Reply:
Fixed the problem at the source, ie. from `OglApp::redisplay()`